plugins/modules/provision_intent.py

```diff
@@ -0,0 +1,522 @@
+
+    def get_wired_params(self):
+
+        wired_params = {
+            "deviceManagementIpAddress": self.validated_config[0]["management_ip_address"],
+            "siteNameHierarchy": self.validated_config[0].get("site_name")
+        }
+
+        return wired_params
+
+    def get_wireless_params(self):
+        wireless_params = [
+            {
+                "site": self.validated_config[0].get("site_name"),
+                "managedAPLocations": self.validated_config[0].get("managed_ap_locations"),
+            }
+        ]
+        for ap_loc in wireless_params[0]["managedAPLocations"]:
+            if self.get_site_type(site_name=ap_loc) != "floor":
+                self.module.fail_json(msg="Managed AP Location must be a floor", response=[])
+
+        wireless_params[0]["dynamicInterfaces"] = []
+        for interface in self.validated_config[0].get("dynamic_interfaces"):
+            interface_dict = {
+                "interfaceIPAddress": interface.get("interface_ip_address"),
+                "interfaceNetmaskInCIDR": interface.get("interface_netmask_in_c_i_d_r"),
+                "interfaceGateway": interface.get("interface_gateway"),
+                "lagOrPortNumber": interface.get("lag_or_port_number"),
+                "vlanId": interface.get("vlan_id"),
+                "interfaceName": interface.get("interface_name")
+            }
+            wireless_params[0]["dynamicInterfaces"].append(interface_dict)
+        response = self.dnac_apply['exec'](
+            family="devices",
+            function='get_network_device_by_ip',
+            params={"management_ip_address": self.validated_config[0]["management_ip_address"]}
+        )
+
+        wireless_params[0]["deviceName"] = response.get("response")[0].get("hostname")
+        return wireless_params
+
+    def get_want(self):
+
+        """
+        Get all provision related informantion from the playbook
+        Args:
+            self: The instance of the class containing the 'config' attribute to be validated.
+            config: validated config passed from the playbook
+        Returns:
+            The method returns an instance of the class with updated attributes:
+                - self.want: A dictionary of paramters obtained from the playbook
+                - self.msg: A message indicating all the paramters from the playbook are
+                collected
+                - self.status: Success
+        Example:
+            It stores all the paramters passed from the playbook for further processing
+            before calling the APIs
+        """
+
+        self.want = {}
+        self.want["device_type"] = self.get_dev_type()
+        if self.want["device_type"] == "wired":
+            self.want["prov_params"] = self.get_wired_params()
+        elif self.want["device_type"] == "wireless":
+            self.want["prov_params"] = self.get_wireless_params()
+        self.msg = "Successfully collected all parameters from playbook " + \
```

Comment:
make something in else.. some log message?

Reply:
Done